cognite/client/data_classes/_base.py

```diff
@@ -260,10 +260,19 @@ def _set_null(self, name):
         self._update_object[name] = {"setNull": True}
 
     def _add(self, name, value):
-        self._update_object[name] = {"add": value}
+        update_obj = self._update_object.get(name, {})
+        assert not ("set" in update_obj), "Can not call remove or add fields after calling set on an update object."
+        if isinstance(value, list):
+            update_obj["add"] = update_obj.get("add", []) + value
+        else:
+            update_obj["add"] = {**update_obj.get("add", {}), **value}
+        self._update_object[name] = update_obj
 
     def _remove(self, name, value):
-        self._update_object[name] = {"remove": value}
+        update_obj = self._update_object.get(name, {})
+        assert not ("set" in update_obj), "Can not call remove or add fields after calling set on an update object."
+        update_obj["remove"] = sorted(list(set(update_obj.get("remove", [])) | set(value)))
```

Comment:
how does this work? not familiar with `|` syntax and for some reason google won't help me 🤔

Reply:
{1,2} | {1,3} == {1,2,3}
set union